gateway-js/src/__tests__/buildQueryPlan.test.ts

```diff
@@ -881,6 +881,7 @@ describe('buildQueryPlan', () => {
                         __typename
                         ... on Image {
                           ... on NamedObject @include(if: $b) {
+                            __typename
```

Comment:
Note to reviewers: there is a handful of cases, like this one, where the change of this commit adds a `__typename` that wasn't there before, even though it's technically redundant (we already ask for the `__typename` of this object a few lines above). Imo, that's fine and it's not worth being bothered by: it doesn't happen much and asking for `__typename` is always pretty cheap, so this probably have no measurable effect. We can try to avoid this kind of duplication later if that bother someone and we have spare cycles, but the check implementation may create more inefficiencies in query planner than it saves at execution.